### PR TITLE
feat(api): /ready probes storage dependency (#291)

### LINKED ISSUE #291: [MEDIUM] GET /ready always reports ready (no dependency probes)
### Summary
`readiness_check` in `crates/api/src/handlers/health.rs` returns `ready: true` and `database: true` unconditionally (TODO ~24). Orchestrators use readiness to avoid routing traffic before dependencies are usable.

### Impact
Traffic can reach the API while storage or other dependencies are down, causing spurious errors under load.

### Suggested direction
Probe `AppState` (repos or a shared health service) with short timeouts; return non-ready / HTTP 503 when dependencies fail.

### References
- `crates/api/src/handlers/health.rs`

### PR DESCRIPTION
**Pull request by vanyastaff**
## Summary

- Closes #291. `GET /ready` used to return `ready: true, database: true`
  unconditionally; orchestrators (k8s `readinessProbe`, load balancers)
  routed traffic even when storage was unreachable.
- `readiness_check` now runs a bounded `workflow_repo.count()` probe under
  a 2 s timeout: success → `200 OK` + `ready: true`; error or timeout →
  `503 Service Unavailable` + `ready: false` + structured `tracing::warn!`
  with the specific failure so operators can diagnose without reading
  source.
- `GET /health` stays dependency-free — it remains a pure liveness signal
  so a dependency outage cannot trigger a restart loop.

## Design notes

Uses the existing `WorkflowRepo::count` surface rather than adding a new
`ping()` trait method — the trait is unchanged, the probe is cheap on
in-memory and adequate on Postgres. A dedicated `ping()` can be added
later if `count()` proves too heavy on very large datasets; the 2 s
timeout guards against that edge case in the meantime.

## Test plan

- [x] `cargo nextest run -p nebula-api handlers::health` — 2/2 pass:
      - `readiness_reports_ok_when_database_responds`
      - `readiness_returns_503_when_database_probe_fails`
- [x] Pre-push lefthook mirror — all green (check-no-default, nextest
      3266 tests, doctests, docs, check-all-features, shear)
- [x] Includes a rustdoc follow-up commit that drops a private
      intra-doc link on `PROBE_TIMEOUT` (matches the repo's
      \`-D rustdoc::private-intra-doc-links\` policy)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Health checks now actively probe system dependencies to determine true readiness status.
  * Added timeout protection (2 seconds) to prevent hanging health checks.
  * Readiness endpoint returns appropriate HTTP status codes (200 for healthy, 503 when unavailable).

* **Tests**
  * Added comprehensive unit tests for health check behavior, including timeout and failure scenarios.

### CODE REVIEW

File: crates/api/src/handlers/health.rs [hunk 1]
```diff
@@ -18,15 +38,138 @@ pub async fn health_check() -> Json<HealthResponse> {
...
+        Ok(Ok(_)) => true,
+        Ok(Err(err)) => {
+            tracing::warn!(error = %err, "readiness probe: workflow repo count() failed");
+            false
+        },
+        Err(_) => {
+            tracing::warn!(
+                timeout_secs = PROBE_TIMEOUT.as_secs(),
+                "readiness probe: workflow repo count() timed out",
+            );
```
[the remark this answers is not in the document]
Comment (vanyastaff) [reply crates/api/src/handlers/health.rs:86]: Added `readiness_returns_503_when_database_probe_times_out` in commit c8ac71cf. Introduces a `SlowWorkflowRepo` whose `count()` sleeps 60s and annotates the test with `#[tokio::test(start_paused = true)]` so tokio's paused-time auto-advance fires `PROBE_TIMEOUT` in virtual time — the test runs in ~20ms rather than 2 real seconds. Asserts `503 Service Unavailable` and `ready: false` on the response body, covering the previously-untested `Err(_)` timeout branch in `probe_database`.
